core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -216,26 +219,23 @@ class DruidBeamMaker[A: Timestamper](
 
 object DruidBeamMaker
 {
-  def generateBaseFirehoseId(dataSource: String, segmentGranularity: Granularity, ts: DateTime, partition: Int) = {
+  def generateBaseFirehoseId(dataSource: String,taskDuration:Period, ts: DateTime, partition: Int) = {
     // Not only is this a nasty hack, it also only works if the RT task hands things off in a timely manner. We'd rather
     // use UUIDs, but this creates a ton of clutter in service discovery.
+    //It may also break if task duration is changed across RT tasks
```

Comment:
ah, I see. This should actually be okay, since tranquility remembers the firehoseId when it creates a task, rather than re-generating it each time. It's stored in the beam metadata in zk and can be restored after app restart.